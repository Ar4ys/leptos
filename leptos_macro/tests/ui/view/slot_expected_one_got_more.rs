use leptos::*;

#[slot]
struct Then {}

#[component]
fn SlotIf(then: Then) -> impl IntoView {
    _ = then;
}

#[component]
fn If(then: Then) -> impl IntoView {
    _ = then;
}

fn main() {
    view! {
        <If>
            <Then slot />
            <Then slot />
        </If>
    };

    view! {
        <SlotIf>
            <Then slot />
            <Then slot />
        </SlotIf>

    };
}
