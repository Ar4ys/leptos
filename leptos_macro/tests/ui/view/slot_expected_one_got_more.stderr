error[E0308]: mismatched types
  --> tests/ui/view/slot_expected_one_got_more.rs:20:13
   |
20 |             <Then slot />
   |             ^^^^^^^^^^^^^
   |             |
   |             expected `Then`, found `Vec<_>`
   |             arguments to this method are incorrect
   |
   = note: expected struct `Then`
              found struct `Vec<_>`
note: method defined here
  --> tests/ui/view/slot_expected_one_got_more.rs:12:7
   |
12 | fn If(then: Then) -> impl IntoView {
   |       ^^^^------
   = note: this error originates in the macro `::std::vec` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> tests/ui/view/slot_expected_one_got_more.rs:27:13
   |
27 |             <Then slot />
   |             ^^^^^^^^^^^^^
   |             |
   |             expected `Then`, found `Vec<_>`
   |             arguments to this method are incorrect
   |
   = note: expected struct `Then`
              found struct `Vec<_>`
note: method defined here
  --> tests/ui/view/slot_expected_one_got_more.rs:7:11
   |
7  | fn SlotIf(then: Then) -> impl IntoView {
   |           ^^^^------
   = note: this error originates in the macro `::std::vec` (in Nightly builds, run with -Z macro-backtrace for more info)
